documentation/src/main/asciidoc/topics/con_file_based_cache_stores.adoc

```diff
@@ -1,72 +1,36 @@
+File-based cache stores should always be unique to each {brandname} instance.
+If you want to use the same persistent across a cluster, configure shared storage, such as a JDBC string-based cache store.
 
-You can configure any number of B+ trees with the `segments` attribute on the `index` element declaratively or with the `indexSegments()` method programmatically.
-By default {brandname} creates up to 16 B+ trees, which means there can be up to 16 indexes.
-Having multiple indexes prevents bottlenecks from concurrent writes to an index and reduces the number of entries that {brandname} needs to keep in memory.
-As it iterates over a soft-index file store, {brandname} reads all entries in an index at the same time.
+The following limitations exist with how a non-shared store interacts in a clustered environment: 
 
-Each entry in the B+ tree is a node.
-By default, the size of each node is limited to 4096 bytes.
-`SoftIndexFileStore` throws an exception if keys are longer after serialization occurs.
+* If you do not gracefully shutdown a cache, only the last node remains active and it must store all the cache's data. In-memory and eviction limitations might impact the amount of data a node can store. When you restart the cache, you can only load data from the last active node. 
+* If you stop a running rebalancing operation during cluster shutdown, {brandname} cannot keep a fixed number of copies of any entry in a cache. When you attempt to restart the cluster, the rebalancing operation remains active and it does not perform checks on nodes that rejoin the cluster. This can lead to corrupted data or data loss for a cache.
 
-.Segmentation
-
-Soft-index file stores are always segmented.
-
-[NOTE]
-====
-The `AdvancedStore.purgeExpired()` method is not implemented in `SoftIndexFileStore`.
+[IMPORTANT]
 ====
+To retain data during clustered cache restarts, use a shared store.
 
-[discrete]
-== Single File Cache Stores
-
-[NOTE]
-====
-Single file cache stores are now deprecated and planned for removal.
+// Downstream content
+ifdef::downstream[]
+See, link:https://access.redhat.com/solutions/6960765[Adding persistence in a cache in RHDG 8 in OCP 4] ({RedHat} Customer Portal).
+endif::downstream[]
 ====
 
-Single File cache stores, `SingleFileStore`, persist data to file.
-{brandname} also maintains an in-memory index of keys while keys and values are stored in the file.
-
-Because `SingleFileStore` keeps an in-memory index of keys and the location of values, it requires additional memory, depending on the key size and the number of keys.
-For this reason, `SingleFileStore` is not recommended for use cases where the keys are larger or there can be a larger number of them.
+If you need to enable persistence for an existing cache on {k8s}, you must delete the cache and then re-create with persistent enabled. 
+Deleting the cache removes all data for the cache, so consider copying any important cache data before you delete the cache.
```

Comment:
This comment applies to all persistence stores, not just file-stores, and in any Infinispan environment not just k8s. We should probably move this up a level to "6. Configuring persistent storage"

Reply:
OK, I moved the paragraph to section 6.0. Should I change the text to reflect shared and non-shared stores in a specific environment?

Reply:
> Should I change the text to reflect shared and non-shared stores in a specific environment?

These docs apply to both types of stores, so need to add any specifc.